api/edge_api/identities/serializers.py

```diff
@@ -77,7 +78,12 @@ def to_internal_value(self, data):
 
 class FeatureStateValueEdgeIdentityField(serializers.Field):
     def to_representation(self, obj):
-        identity_id = self.parent.get_identity_uuid()
+        identity: EdgeIdentity = self.parent.context["identity"]
+        environment: Environment = self.parent.context["environment"]
+        identity_id = identity.get_hash_key(
+            environment.use_identity_composite_key_for_hashing
+        )
+
```

Comment:
This is technically out of the scope of this PR, but this was an issue in the current implementation of the views which retrieve the feature states for a given identity.

The previous logic, although wrong, wouldn't have actually given any incorrect data (at least for users that are creating identity overrides via the dashboard, not via the API) since it only would have affected multivariate features with more than one weighted option, which we don't allow for identity overrides (you have to select a single option, which will be given 100% weighting). 

Since I had to also implement this functionality for the new endpoint, however, it made sense to fix it in this PR.